modules/gplazma2-jython/src/main/java/org/dcache/gplazma/jython/JythonPlugin.java

```diff
@@ -0,0 +1,175 @@
+package org.dcache.gplazma.jython;
+
+import java.io.IOException;
+import java.nio.file.DirectoryStream;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.security.Principal;
+import java.util.Set;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Properties;
+import java.util.stream.Collectors;
+
+import org.dcache.auth.Subjects;
+import org.dcache.gplazma.AuthenticationException;
+import org.dcache.gplazma.plugins.GPlazmaAuthenticationPlugin;
+import org.python.core.PyFunction;
+import org.python.core.PyObject;
+import org.python.core.PySet;
+import org.python.util.PythonInterpreter;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+
+public class JythonPlugin implements GPlazmaAuthenticationPlugin {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(JythonPlugin.class);
+
+    private final PythonInterpreter PI = new PythonInterpreter();
+    private List<Path> pluginPathList = new ArrayList<>();
+
+    public JythonPlugin(Properties properties) {
+        // constructor
+        try {
+            pluginPathList = listDirectoryContents("gplazma2-jython");  // TODO fix this, currently just reads from local folder
+        } catch (IOException e) {
+            LOGGER.error("Error reading Jython directory");
+        }
+    }
+
+    private static List<Path> listDirectoryContents(String directoryPath) throws IOException {
+        List<Path> elements = new ArrayList<>();
+        Path path = Paths.get(directoryPath);
+
+        // Check if the path is a directory
+        if (Files.isDirectory(path)) {
+            // Use DirectoryStream to iterate through the directory
+            try (DirectoryStream<Path> directoryStream = Files.newDirectoryStream(path)) {
+                for (Path p : directoryStream) {
+                    elements.add(p);
+                }
+            }
+        } else {
+            throw new IllegalArgumentException("Path is not a directory: " + directoryPath);
+        }
+
+        return elements;
+    }
+
+    public void authenticate(
+            Set<Object> publicCredentials,
+            Set<Object> privateCredentials,
+            Set<Principal> principals
+    ) throws AuthenticationException {
+
+        for (Path pluginPath : pluginPathList) {
+            // for each path in pluginPathList
+            try {
+                // ===========
+                // IO
+                // ===========
+                // read file into string
+                LOGGER.debug("gplazma2-jython running " + pluginPath.toString());
+                String content = Files.lines(
+                        Paths.get(pluginPath.toString())
```

Comment:
`pluginPath` is already `Path`. Can't we use it directly: `Files.lines(pluginPath)`?